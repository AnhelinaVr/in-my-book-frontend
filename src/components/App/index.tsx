import Button from '@mui/material/Button';
import React from 'react';
import EditIcon from '@mui/icons-material/Edit';
import CloseIcon from '@mui/icons-material/Close';
import AddIcon from '@mui/icons-material/Add';
import DeleteIcon from '@mui/icons-material/Delete';
import SearchIcon from '@mui/icons-material/Search';

const App = () => {
  return (
    <>
      <h1>Start Project</h1>
      <Button variant="contained">Create account</Button>
      <Button variant="outlined">Outlined</Button>
      <Button variant="outlined">View</Button>
      <Button variant="outlined" startIcon={<SearchIcon />}>
        Search
      </Button>
      <Button variant="outlined" startIcon={<EditIcon />}>
        Edit
      </Button>
      <Button variant="outlined" startIcon={<DeleteIcon />}>
        Delete
      </Button>
      <Button variant="outlined" startIcon={<AddIcon />}>
        Add
      </Button>
      <Button variant="outlined" startIcon={<CloseIcon />}>
        Decline
      </Button>
    </>
  );
};

export default App;
