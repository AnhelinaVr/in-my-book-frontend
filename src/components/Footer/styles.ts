import { SxProps, Theme } from '@mui/material';

export const getStyles = (theme: Theme): Record<string, SxProps> => ({
  container: {
    width: '100%',
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    background: theme.palette.background.darker,
    padding: theme.spacing(5),
  },
  github: {
    display: 'flex',
    alignItems: 'center',
    marginBottom: theme.spacing(2),
    fontWeight: theme.typography.fontWeightBold,
  },
  githubLink: {
    textDecoration: 'none',
    fontSize: 12,
    marginLeft: theme.spacing(1),
    '&:hover': {
      textDecoration: 'underline',
    },
  },
});
