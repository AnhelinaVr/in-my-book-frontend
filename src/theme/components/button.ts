export const MuiButton = {
  styleOverrides: {
    root: {
      borderRadius: 20,
      fontWeight: 'bold',
      textTransform: 'capitalize' as const,
      padding: '4px 20px',
    },
  },
};
