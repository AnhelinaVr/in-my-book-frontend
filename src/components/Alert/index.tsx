import { Alert as MuiAlert, Snackbar } from '@mui/material';
import { Theme, useTheme } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import { AlertType } from '../../types';
import { AppState } from '../../redux/rootReducer';
import ErrorIcon from '@mui/icons-material/WarningRounded';
import SuccessIcon from '@mui/icons-material/CheckRounded';
import { getStyles } from './styles';
import { hideAlert } from '../../redux/alert/actions';

const getIcon = (type: AlertType, theme: Theme) =>
  type === AlertType.SUCCESS ? (
    <SuccessIcon sx={{ color: theme.palette.success.dark }} />
  ) : (
    <ErrorIcon sx={{ color: theme.palette.warning.main }} />
  );

export const Alert = () => {
  const dispatch = useDispatch();
  const theme = useTheme();
  const styles = getStyles(theme);

  const alert = useSelector((state: AppState) => state.alert);

  const handleClose = (
    _event?: Event | React.SyntheticEvent,
    reason?: string,
  ) => {
    if (reason === 'clickaway') {
      return;
    }

    dispatch(hideAlert());
  };

  return (
    <>
      {alert && (
        <Snackbar
          open={alert.isOpen}
          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
          autoHideDuration={alert.timeout}
          onClose={handleClose}
        >
          <MuiAlert
            severity={alert.type}
            onClose={handleClose}
            sx={() => ({ ...styles.root, ...styles[alert.type] })}
            icon={getIcon(alert.type, theme)}
          >
            {alert.message}
          </MuiAlert>
        </Snackbar>
      )}
    </>
  );
};
