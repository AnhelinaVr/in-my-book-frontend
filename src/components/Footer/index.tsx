import { Box, Link, Typography, useTheme } from '@mui/material';
import { ReactComponent as GithubIcon } from './images/gh-icon.svg';
import { getStyles } from './styles';

export const Footer = () => {
  const theme = useTheme();
  const styles = getStyles(theme);

  return (
    <Box sx={styles.container}>
      <Box sx={styles.github}>
        <GithubIcon />
        <Link
          href="https://github.com/in-my-book-team"
          target="_blank"
          sx={styles.githubLink}
        >
          GitHub
        </Link>
      </Box>
      <Typography
        sx={{ fontSize: 12, fontWeight: theme.typography.fontWeightBold }}
      >
        Copyright Â© 2023. All rights reserved
      </Typography>
    </Box>
  );
};
