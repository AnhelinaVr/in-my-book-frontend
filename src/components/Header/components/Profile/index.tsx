import { Box, Button, useTheme } from '@mui/material';
import { getStyles } from './styles';

export const Profile = () => {
  const theme = useTheme();
  const styles = getStyles(theme);

  return (
    <Box sx={styles.profileContainer} data-testid="header-profile">
      <Button>Username</Button>
    </Box>
  );
};
