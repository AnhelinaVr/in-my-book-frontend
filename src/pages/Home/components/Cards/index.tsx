import { Box, Button, Typography, useTheme } from '@mui/material';
import { CSSProperties } from 'react';
import { getStyles } from './styles';
import { useNavigate } from 'react-router-dom';

type Card = {
  title?: string;
  subtitle?: string;
  image: string;
  description: string;
  button?: { text: string; path: string };
};

type Props = {
  cards: Card[];
};

export const Cards: React.FC<Props> = ({ cards }) => {
  const theme = useTheme();
  const styles = getStyles(theme);
  const navigate = useNavigate();

  const onClick = (path = '/') => {
    navigate(path);
  };

  return (
    <>
      <Box sx={styles.container}>
        {cards.map((card) => (
          <Box sx={styles.card}>
            {card.title && (
              <Typography variant="h3" sx={styles.title}>
                {card.title}
              </Typography>
            )}
            <img
              style={
                (card.subtitle
                  ? styles.borderedImage
                  : styles.image) as CSSProperties
              }
              src={card.image}
              alt="card"
            />
            {card.subtitle && (
              <Typography variant="h4" sx={styles.subtitle}>
                {card.subtitle}
              </Typography>
            )}
            <Typography sx={styles.description}>{card.description}</Typography>
            {card.button && (
              <Button
                onClick={() => onClick(card.button?.path)}
                sx={styles.button}
              >
                {card.button.text}
              </Button>
            )}
          </Box>
        ))}
      </Box>
    </>
  );
};
