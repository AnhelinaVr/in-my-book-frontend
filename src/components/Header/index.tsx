import { Box, Button, Link, useTheme } from '@mui/material';
import { Link as LinkType } from '../../types';
import { ReactComponent as Logo } from './images/Logo.svg';
import { Paths } from '../../constants';
import { Profile } from './components/Profile';
import React from 'react';
import { getStyles } from './styles';
import { useNavigate } from 'react-router-dom';

type Props = {
  isAuthenticated: boolean;
  links: LinkType[];
};

export const Header: React.FC<Props> = ({ isAuthenticated, links }) => {
  const theme = useTheme();
  const styles = getStyles(theme);
  const navigate = useNavigate();

  const registrationOnClick = (path: LinkType['path']) => {
    navigate(path);
  };

  return (
    <Box sx={styles.container}>
      <Box sx={styles.logoContainer}>
        <Logo />
      </Box>
      <Box sx={styles.links}>
        {links.map((link) => (
          <Link href={link.path} key={link.name} sx={styles.link}>
            {link.name}
          </Link>
        ))}
      </Box>
      {isAuthenticated ? (
        <Profile />
      ) : (
        <Box sx={styles.registrationContainer}>
          {/* TODO: button hovers */}
          <Button onClick={() => registrationOnClick(Paths.home)}>
            Sign In
          </Button>
          <Button
            onClick={() => registrationOnClick(Paths.home)}
            sx={styles.signUp}
          >
            Sign Up
          </Button>
        </Box>
      )}
    </Box>
  );
};
