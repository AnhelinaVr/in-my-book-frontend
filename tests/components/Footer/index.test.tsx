import { Footer } from '../../../src/components/Footer';
import React from 'react';
import { render } from '@testing-library/react';

describe('<Home />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
