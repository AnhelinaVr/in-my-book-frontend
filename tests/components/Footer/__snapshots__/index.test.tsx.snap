// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1e5i6q7"
  >
    <div
      class="MuiBox-root css-1mioigq"
    >
      <img
        alt="Github Logo"
        src="function SvgGhIcon(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgGhIcon.name})
  );
}"
      />
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-j3poeh-MuiTypography-root-MuiLink-root"
        href="https://github.com/in-my-book-team"
        target="_blank"
      >
        GitHub
      </a>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1jx4n4e-MuiTypography-root"
    >
      Copyright Â© 2023. All rights reserved
    </p>
  </div>
</div>
`;
